File: 1351-count-negative-numbers-in-a-sorted-matrix/Solution.cs
using System.Runtime.InteropServices;

public class Solution {
    public int CountNegatives(int[][] grid)
    {
        int negatives = 0;
        for (int i = 0; i < grid.Length; i++)
        {
            int index = SearchInsert(grid[i], -1);
            negatives += grid[i].Length - index;
        }
        return negatives;
    }
    private int SearchInsert(int[] nums, int target)
    {
        var leftIndex = 0;
        var rightIndex = nums.Length - 1;
        while (leftIndex <= rightIndex)
        {
            var middleIndex = (leftIndex + rightIndex) / 2;
            if (nums[middleIndex] <= target)
            {
                rightIndex = middleIndex - 1;
            }
            else
            {
                leftIndex = middleIndex + 1;
            }
        }

        return leftIndex;
    }
}

File: 1351-count-negative-numbers-in-a-sorted-matrix/Solution.go
package Leetcode

func countNegatives(grid [][]int) int {
	var negatives = 0
	for i := 0; i < len(grid); i++ {
		var index = searchInsert(grid[i], -1)
		negatives += len(grid[i]) - index
	}
	return negatives

}

func searchInsert(nums []int, target int) int {
	var left = 0
	var right = len(nums) - 1
	for left <= right {
		var mid = (left + right) / 2
		if nums[mid] <= target {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}
	return left
}
